pkg/api/obligationmap.go

```diff
@@ -12,38 +12,51 @@ import (
 	"time"
 
 	"golang.org/x/exp/slices"
+	"gorm.io/gorm"
 
 	"github.com/fossology/LicenseDb/pkg/db"
 	"github.com/fossology/LicenseDb/pkg/models"
 	"github.com/fossology/LicenseDb/pkg/utils"
 	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
 )
 
-// GetObligationMapByTopic retrieves obligation maps for a given obligation topic
+// GetObligationMapByObligationId retrieves obligation maps for a given obligation id
 //
 //	@Summary		Get maps for an obligation
-//	@Description	Get obligation maps for a given obligation topic
-//	@Id				GetObligationMapByTopic
+//	@Description	Get obligation maps for a given obligation id
+//	@Id				GetObligationMapByObligationId
 //	@Tags			Obligations
 //	@Accept			json
 //	@Produce		json
-//	@Param			topic	path		string	true	"Topic of the obligation"
-//	@Success		200		{object}	models.ObligationMapResponse
-//	@Failure		404		{object}	models.LicenseError	"No obligation with given topic found or no map for
-//	obligation exists"
+//	@Param			id	path		string	true	"Id of the obligation"
+//	@Success		200	{object}	models.ObligationMapResponse
+//	@Failure		404	{object}	models.LicenseError	"No obligation with given id found"
 //	@Security		ApiKeyAuth || {}
-//	@Router			/obligation_maps/topic/{topic} [get]
-func GetObligationMapByTopic(c *gin.Context) {
+//	@Router			/obligation_maps/obligation/{id} [get]
+func GetObligationMapByObligationId(c *gin.Context) {
 	var obligation models.Obligation
 	var resObMap models.ObligationMapUser
-	var shortnameList []string
+	var list []models.ObligationMapLicenseFormat
 
-	topic := c.Param("topic")
+	queryParam := c.Param("id")
+	id, err := uuid.Parse(queryParam)
```

Comment:
```suggestion
	id, err := uuid.Parse(c.Param("id"))
```

Can this be written like this? also `id` => `obligationId`???